components/shared/ui/menu.tsx

```diff
@@ -1,53 +1,99 @@
+        border-gray-500 border-l-8 border-t-0 md:border-t-8 md:border-l-0 h-full ${mainColorBgClass[siteCodename]} group grow`} onClick={() => props.handleClick(i)}>
+          {link.elements.subitems.value.length > 0 ? (
+            <div className={`${i === props.activeMenu ? "bg-white " : ""} md:hover:bg-white h-full`}>
+              <DropdownButton item={link} />
+              <div
+                className={`${i === props.activeMenu ? "block" : "hidden"} md:group-hover:block absolute z-50 left-0 shadow-sm bg-white border-gray-200 w-full`}
+              >
+                <div className="grid gap-2 max-w-screen-xl px-4 py-5 mx-auto text-gray-900 sm:grid-cols-2 md:grid-cols-3 md:px-6">
+                  <DropdownMenuItems links={link.elements.subitems.linkedItems} />
+                </div>
+              </div>
+            </div>
+          ) : (
+            <Link
+              {...link.elements.openInANewWindow.value[0] ? { rel: "noopener noreferrer", target: "_blank" } : {}}
+              className={`h-full flex items-center justify-between w-full py-2 pl-3 pr-4 font-medium text-gray-900 border-b border-gray-100 md:w-auto md:bg-transparent md:border-0 md:hover:bg-white`}
+              href={link.elements.externalLink.value ? link.elements.externalLink.value : "/" + link.elements.pageLink.linkedItems[0].elements.url.value}
+            >
+              {link.elements.label.value}
+            </Link>
+          )}
+        </li>
       ))}
     </>
-  )
+  );
 }
 
 const DropdownButton: FC<Props> = props => {
   return (
     <button
-      className="h-full flex items-center justify-between w-full py-2 pl-3 pr-4 font-medium text-gray-900 border-b border-gray-100 md:w-auto md:bg-transparent md:border-0"
+      className={`h-full flex items-center justify-between w-full p-4 py-2 font-medium text-gray-900 border-b border-gray-100 md:w-auto md:bg-transparent md:border-0`}
     >
       {props.item.elements.label.value}
       <ChevronDownIcon className="w-4 h-4 ml-1 mt-1" />
     </button>
   )
 }
 
+const DropdownMenuItems: FC<DropdownMenuProps> = props => {
+  const siteCodename = useSiteCodename();
+  const router = useRouter();
+
+  return (
+    <>
+      {props.links.map(link => (
+        <Link
+          key={link.system.codename}
+          {...link.elements.openInANewWindow.value[0] ? { rel: "noopener noreferrer", target: "_blank" } : {}}
+          href={link.elements.externalLink.value ? link.elements.externalLink.value : "/" + link.elements.pageLink.linkedItems[0].elements.url.value}
+          className={`${"/" + link.elements.pageLink.linkedItems[0]?.elements.url.value === router.asPath ? "border-l-gray-500 cursor-default" : "border-l-transparent"}
+          block p-3 bg-gray-200 border-l-transparent border-l-8 hover:border-l-8 hover:border-l-gray-500`}
```

Comment:
I don't think hover:border-l-8 is necessary

Reply:
NITPICK: I like bg-gray-100 and hover:border-l-gray-300 as it is not so in contrast with the pastel colors of the menu. What do you think?

Reply:
I kept it as it was - lighthouse does not showed any problem in contrast.